src/main/java/htsjdk/samtools/util/IntervalList.java

```diff
@@ -869,4 +844,68 @@ public int hashCode() {
         result = 31 * result + intervals.hashCode();
         return result;
     }
+
+    public static class IntervalMerger implements Iterable<Interval> {
+
+        Iterator<Interval> inputIntervals;
+        Interval current = null;
+        final boolean combineAbuttingIntervals;
+        final boolean enforceSameStrands;
+        final boolean concatenateNames;
+        final List<Interval> toBeMerged = new ArrayList<>();
+
+        public IntervalMerger(Iterator<Interval> intervals, final boolean combineAbuttingIntervals, final boolean enforceSameStrand, final boolean concatenateNames) {
+            this.inputIntervals = intervals;
+
+            this.combineAbuttingIntervals = combineAbuttingIntervals;
+            this.enforceSameStrands = enforceSameStrand;
+            this.concatenateNames = concatenateNames;
+        }
+
+        @Override
+        public Iterator<Interval> iterator() {
+
+            return new Iterator<Interval>() {
+                @Override
+                public boolean hasNext() {
+                    return current != null || inputIntervals.hasNext();
+                }
+
+                @Override
+                public Interval next() {
+                    return getNext();
+                }
+
+                private Interval getNext() {
+                    Interval next = null;
```

Comment:
Shouldn't this throw if hasNext() isn't true?

Reply:
oops. thanks.